core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataServiceSpec.js

```diff
@@ -305,48 +226,37 @@ describe('TrainingDataService', function() {
 
     // Verify initial state.
     var state = ess.getState('State');
-    expect(state.interaction.answerGroups[0].rules[1]).toEqual(
-      rof.createNew(RULE_TYPE_CLASSIFIER, {
-        training_data: ['text answer']
-      })
-    );
+    expect(state.interaction.answerGroups[0].training_data).toEqual([
+      'text answer'
+    ]);
     expect(state.interaction.confirmedUnclassifiedAnswers).toEqual([
       'second answer'
     ]);
 
     // Training a duplicate answer for the answer group should change nothing.
     tds.trainAnswerGroup(0, 'text answer');
     state = ess.getState('State');
-    expect(state.interaction.answerGroups[0].rules[1]).toEqual(
-      rof.createNew(RULE_TYPE_CLASSIFIER, {
-        training_data: ['text answer']
-      })
-    );
+    expect(state.interaction.answerGroups[0].training_data).toEqual([
+      'text answer'
+    ]);
 
     // Training a duplicate answer for the default response should change
     // nothing.
     tds.trainDefaultResponse('second answer');
     state = ess.getState('State');
-    expect(state.interaction.answerGroups[0].rules[1]).toEqual(
-      rof.createNew(RULE_TYPE_CLASSIFIER, {
-        training_data: ['text answer']
-      })
-    );
+    expect(state.interaction.answerGroups[0].training_data).toEqual([
+      'text answer'
+    ]);
   });
 
   it('should remove unresolved answers after training', function() {
-    tds.initializeTrainingData('0', 'State');
-    $httpBackend.flush();
-
     // Training an answer group should remove an unresolved answer.
     tds.trainAnswerGroup(0, 'answer1');
-    expect(tds.getTrainingDataAnswers()).toEqual(['answer2']);
-    expect(tds.getTrainingDataFrequencies()).toEqual([1]);
+    expect(tds.getTrainingDataAnswers()).toEqual([]);
 
     // Training the default response should also remove an answer.
     tds.trainDefaultResponse('answer2');
     expect(tds.getTrainingDataAnswers()).toEqual([]);
-    expect(tds.getTrainingDataFrequencies()).toEqual([]);
```

Comment:
Do we still need getTrainingDataFrequencies()? The tests that use it seem to have been removed in this PR.

Reply:
OK, let's remove it if we don't need it then. I think the StateAnswersModel has evolved since this was written.

Reply:
Done